R/travis-meta.R

```diff
@@ -44,13 +50,20 @@ format.travis_repo <- function(x, ..., short = FALSE) {
 #' @description
 #' `travis_user()` queries the "/users" API.
 #'
+#' @template quiet
 #' @export
 #'
 #' @rdname travis_repos
-travis_user <- function(token = travis_token()) {
-  req <- TRAVIS_GET3("/user", token = token)
-  httr::stop_for_status(req, paste("get current user information"))
-  new_travis_user(httr::content(req))
+travis_user <- function(quiet = FALSE) {
+
+  req <- travis(path = "/user")
+
+  if (status_code(req$response) == 200) {
```

Comment:
Stopping for status?

I see quite a few instances of `== 200` in this PR, not tagging every single one.

Reply:
Yeah, adding `stop_for_status(req$response)` and removing all the status code conditions makes sense. And we can have proper error messages for certain requests.

Reply:
I've added `stop_for_status()` with meaningful error msgs to all funs.